File: java/spring-resilience4j/spring-resilience4j-api-a/src/main/java/com/bmarques/springresilience4japia/invoice/InvoiceClient.java
package com.bmarques.springresilience4japia.invoice;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Repository;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.List;

@Repository
@FeignClient(name = "invoiceservice", url = "${api.message.center.url}")
public interface InvoiceClient {
    @GetMapping(value = "/invoice", consumes = MediaType.APPLICATION_JSON_VALUE)
    List<Invoice> getInvoices();
}

File: kotlin/spring-resilience4j/spring-resilience4j-api-a/src/main/kotlin/com/bmarques/springresilience4japia/invoice/InvoiceClient.kt
package com.bmarques.springresilience4japia.invoice

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.MediaType
import org.springframework.stereotype.Repository
import org.springframework.web.bind.annotation.GetMapping

@Repository
@FeignClient(name = "invoiceservice", url = "\${api.message.center.url}")
interface InvoiceClient {
    @GetMapping(value = ["/invoice"], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun getInvoices(): List<Invoice>
}